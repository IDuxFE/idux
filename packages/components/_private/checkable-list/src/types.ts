/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { VirtualScrollToFn } from '@idux/cdk/scroll'
import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray, VKey } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export interface CheckableListData {
  key?: VKey
  label?: string
  disabled?: boolean
  [key: string]: unknown
}

export interface CheckableListScroll {
  height?: string | number
  fullHeight?: boolean
}

export const checkableListProps = {
  dataSource: Array as PropType<CheckableListData[]>,
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  checked: Function as PropType<(item: CheckableListData) => boolean>,
  customAdditional: { type: Function as PropType<CheckableListCustomAdditional>, default: undefined },
  disabled: Function as PropType<(item: CheckableListData) => boolean>,
  getKey: Function as PropType<(item: CheckableListData) => VKey>,
  labelKey: String,
  virtual: {
    type: Boolean,
    default: false,
  },
  scroll: Object as PropType<CheckableListScroll>,
  onCheckChange: [Function, Array] as PropType<MaybeArray<(item: CheckableListData, checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<(item: CheckableListData) => void>>,
  onScroll: [Function, Array] as PropType<MaybeArray<(evt: Event) => void>>,
  onScrolledChange: [Function, Array] as PropType<
    MaybeArray<(startIndex: number, endIndex: number, visibleData: unknown[]) => void>
  >,
  onScrolledBottom: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export const checkableListItemProps = {
  checked: {
    type: Boolean,
    default: false,
  },
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  label: String,
  value: {
    type: [String, Number, Symbol] as PropType<VKey>,
    required: true,
  },
  onCheckChange: [Function, Array] as PropType<MaybeArray<(checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export interface CheckableListApi {
  scrollTo: VirtualScrollToFn
}

export type CheckableListProps = ExtractInnerPropTypes<typeof checkableListProps>
export type CheckableListItemProps = ExtractInnerPropTypes<typeof checkableListItemProps>
export type CheckableListPublicProps = ExtractPublicPropTypes<typeof checkableListProps>
export type CheckableListComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CheckableListPublicProps> & CheckableListPublicProps,
  CheckableListApi
>
export type CheckableListInstance = InstanceType<DefineComponent<CheckableListProps, CheckableListApi>>

export type CheckableListCustomAdditional = (options: {
  data: CheckableListData
  index: number
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
}) => Record<string, any> | undefined
