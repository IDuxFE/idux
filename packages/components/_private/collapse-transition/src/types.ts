/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes } from 'vue'

import { IxPropTypes } from '@idux/cdk/utils'

export type CollapseTransitionMode = 'height' | 'width'

export const collapseTransitionProps = {
  appear: IxPropTypes.bool.def(false),
  name: IxPropTypes.string,
  mode: IxPropTypes.oneOf<CollapseTransitionMode>(['height', 'width']).def('height'),
  onAfterEnter: IxPropTypes.emit(),
  onAfterLeave: IxPropTypes.emit(),
}

export type CollapseTransitionProps = ExtractInnerPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionPublicProps = ExtractPublicPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CollapseTransitionPublicProps> & CollapseTransitionPublicProps
>
export type CollapseTransitionInstance = InstanceType<DefineComponent<CollapseTransitionProps>>
