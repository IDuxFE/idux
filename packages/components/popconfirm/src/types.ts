/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes, VKey } from '@idux/cdk/utils'
import type { ButtonProps } from '@idux/components/button'
import type { DefineComponent, HTMLAttributes, VNode } from 'vue'

import { IxPropTypes } from '@idux/cdk/utils'
import { ɵTooltipProps } from '@idux/components/tooltip'

export const popconfirmProps = {
  ...ɵTooltipProps,
  cancelButton: IxPropTypes.object<ButtonProps>(),
  cancelText: IxPropTypes.string,
  okButton: IxPropTypes.object<ButtonProps>(),
  okText: IxPropTypes.string,
  icon: IxPropTypes.string.def('exclamation-circle-filled'),
  footer: IxPropTypes.oneOfType([Boolean, IxPropTypes.array<PopconfirmButtonProps>(), IxPropTypes.vNode]).def(true),

  'onUpdate:visible': IxPropTypes.emit<(visible: boolean) => void>(),
  onCancel: IxPropTypes.emit<(evt?: Event | unknown) => unknown>(),
  onOk: IxPropTypes.emit<(evt?: Event | unknown) => unknown>(),
}
export interface PopconfirmButtonProps<K = VKey> extends ButtonProps {
  key?: K
  text?: string | VNode
  onClick?: (evt: Event) => void
}

export type PopconfirmProps = ExtractInnerPropTypes<typeof popconfirmProps>
export type PopconfirmPublicProps = ExtractPublicPropTypes<typeof popconfirmProps>
export interface PopconfirmBindings {
  updatePopper: () => void
  cancel: (evt?: Event | unknown) => Promise<void>
  ok: (evt?: Event | unknown) => Promise<void>
}
export type PopconfirmComponent = DefineComponent<
  Omit<HTMLAttributes, keyof PopconfirmPublicProps> & PopconfirmPublicProps,
  PopconfirmBindings
>
export type PopconfirmInstance = InstanceType<DefineComponent<PopconfirmProps, PopconfirmBindings>>
