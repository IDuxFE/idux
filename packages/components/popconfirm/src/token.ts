/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { PopconfirmProps } from './types'
import type { Locale } from '@idux/components/locales'
import type { ComputedRef, InjectionKey, Ref } from 'vue'

export interface PopconfirmContext {
  props: PopconfirmProps
  locale: Locale
  mergedPrefixCls: ComputedRef<string>
  visible: ComputedRef<boolean>
  cancelLoading: Ref<boolean>
  okLoading: Ref<boolean>
  cancel: (evt?: unknown) => Promise<void>
  ok: (evt?: unknown) => Promise<void>
}

export const popconfirmToken: InjectionKey<PopconfirmContext> = Symbol('popconfirmToken')
