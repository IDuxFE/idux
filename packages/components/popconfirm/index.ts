/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { PopconfirmComponent } from './src/types'

import Popconfirm from './src/Popconfirm'

const IxPopconfirm = Popconfirm as unknown as PopconfirmComponent

export type { PopconfirmInstance, PopconfirmPublicProps as PopconfirmProps } from './src/types'

export { IxPopconfirm }

export type { PopconfirmButtonProps, PopconfirmBindings } from './src/types'
