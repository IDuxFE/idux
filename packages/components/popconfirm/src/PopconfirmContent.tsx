/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import { computed, defineComponent, inject } from 'vue'

import { ɵFooter } from '@idux/components/_private/footer'
import { IxIcon } from '@idux/components/icon'

import { popconfirmToken } from './token'
import { type PopconfirmButtonProps } from './types'

export default defineComponent({
  setup() {
    const { props, slots, locale, mergedPrefixCls, cancelLoading, okLoading, cancel, ok } = inject(popconfirmToken)!

    const cancelText = computed(() => props.cancelText ?? locale.popconfirm.cancelText)
    const okText = computed(() => props.okText ?? locale.popconfirm.okText)

    const cancelButton = computed<PopconfirmButtonProps>(() => {
      return { size: 'xs', ...props.cancelButton }
    })

    const okButton = computed<PopconfirmButtonProps>(() => {
      return { size: 'xs', ...props.okButton }
    })

    return () => {
      const prefixCls = mergedPrefixCls.value

      return (
        <div class={`${prefixCls}-wrapper`}>
          <div class={`${prefixCls}-title`}>
            {slots.icon ? slots.icon() : <IxIcon name={props.icon}></IxIcon>}
            <span>{slots.title ? slots.title() : props.title}</span>
          </div>
          <ɵFooter
            v-slots={slots}
            class={`${prefixCls}-footer`}
            cancel={cancel}
            cancelButton={cancelButton.value}
            cancelLoading={cancelLoading.value}
            cancelText={cancelText.value}
            footer={props.footer}
            ok={ok}
            okButton={okButton.value}
            okLoading={okLoading.value}
            okText={okText.value}
          />
        </div>
      )
    }
  },
})
