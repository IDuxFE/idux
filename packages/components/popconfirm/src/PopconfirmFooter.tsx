/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { PopconfirmButtonProps } from './types'
import type { VNodeTypes } from 'vue'

import { computed, defineComponent, inject, isVNode } from 'vue'

import { IxButton } from '@idux/components/button'
import { getLocale } from '@idux/components/i18n'

import { popconfirmToken } from './token'

export default defineComponent({
  setup() {
    const { props, slots, mergedPrefixCls, cancelLoading, okLoading, cancel, ok } = inject(popconfirmToken)!
    const locale = getLocale('popconfirm')

    const cancelText = computed(() => props.cancelText ?? locale.value.cancelText)
    const okText = computed(() => props.okText ?? locale.value.okText)

    const okButtonProps = computed<PopconfirmButtonProps>(() => {
      return {
        text: okText.value,
        onClick: ok,
        loading: okLoading.value,
        mode: 'primary',
        ...props.okButton,
      }
    })

    const cancelButtonProps = computed<PopconfirmButtonProps>(() => {
      return {
        text: cancelText.value,
        onClick: cancel,
        loading: cancelLoading.value,
        ...props.cancelButton,
      }
    })

    return () => {
      const { footer } = props
      if (footer === false) {
        return null
      }

      let children: VNodeTypes
      if (slots.footer) {
        children = slots.footer({ cancel, cancelLoading, ok, okLoading })
      } else if (isVNode(footer)) {
        children = footer
      } else {
        let buttonProps = footer
        if (!Array.isArray(buttonProps)) {
          buttonProps = [cancelButtonProps.value, okButtonProps.value]
        }
        children = buttonProps.map(item => {
          const { text, ...rest } = item
          return <IxButton {...rest}>{text}</IxButton>
        })
      }

      return <div class={`${mergedPrefixCls.value}-footer`}>{children}</div>
    }
  },
})
