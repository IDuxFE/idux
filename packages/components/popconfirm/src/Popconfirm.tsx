/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { PopconfirmProps } from './types'

import { computed, defineComponent, provide, ref } from 'vue'

import { callEmit, isPromise } from '@idux/cdk/utils'
import { ɵOverlay } from '@idux/components/_private'
import { useGlobalConfig } from '@idux/components/config'
import { ɵUseTooltipOverlay } from '@idux/components/tooltip'

import PopconfirmContent from './PopconfirmContent'
import { popconfirmToken } from './token'
import { popconfirmProps } from './types'

export default defineComponent({
  name: 'IxPopconfirm',
  props: popconfirmProps,
  setup(props, { slots, expose }) {
    const common = useGlobalConfig('common')
    const mergedPrefixCls = computed(() => `${common.prefixCls}-popconfirm`)
    const config = useGlobalConfig('popconfirm')

    const { overlayRef, updatePopper, visible, setVisible, overlayProps } = ɵUseTooltipOverlay(
      props,
      config,
      mergedPrefixCls,
    )

    const { cancelLoading, okLoading, cancel, ok } = useTrigger(props, setVisible)

    provide(popconfirmToken, {
      props,
      slots,
      mergedPrefixCls,
      visible,
      cancelLoading,
      okLoading,
      cancel,
      ok,
    })

    expose({ updatePopper, cancel, ok })

    return () => {
      const prefixCls = mergedPrefixCls.value
      return (
        <ɵOverlay
          ref={overlayRef}
          v-slots={{ default: slots.default, content: () => <PopconfirmContent /> }}
          class={prefixCls}
          transitionName={`${common.prefixCls}-fade`}
          {...overlayProps.value}
        />
      )
    }
  },
})

function useTrigger(props: PopconfirmProps, setVisible: (value: boolean) => void) {
  const cancelLoading = ref(false)
  const cancel = async (evt?: Event | unknown) => {
    let result = callEmit(props.onCancel, evt)
    if (isPromise(result)) {
      cancelLoading.value = true
      result = await result
      cancelLoading.value = false
    }
    if (result === false) {
      return
    }
    setVisible(false)
  }

  const okLoading = ref(false)
  const ok = async (evt?: Event | unknown) => {
    let result = callEmit(props.onOk, evt)
    if (isPromise(result)) {
      okLoading.value = true
      result = await result
      okLoading.value = false
    }
    if (result === false) {
      return
    }
    setVisible(false)
  }

  return { cancelLoading, okLoading, cancel, ok }
}
