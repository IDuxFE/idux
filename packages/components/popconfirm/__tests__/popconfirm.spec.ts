import type { PopconfirmProps } from '../src/types'
import type { MountingOptions } from '@vue/test-utils'

import { VueWrapper, flushPromises, mount } from '@vue/test-utils'
import { h } from 'vue'

import { renderWork } from '@tests'

import { IxButton } from '@idux/components/button'

import Popconfirm from '../src/Popconfirm'
import PopconfirmContent from '../src/PopconfirmContent'
import { PopconfirmButtonProps, PopconfirmInstance } from '../src/types'

describe('Popconfirm', () => {
  const content = [h('p', 'Some contents...')]
  const PopconfirmMount = (options?: MountingOptions<Partial<PopconfirmProps>>) => {
    const { props, slots, ...rest } = options || {}
    const _options = {
      props: { visible: true, ...props },
      slots: { default: content, ...slots },
      attachTo: 'body',
      ...rest,
    } as MountingOptions<PopconfirmProps>
    return mount(Popconfirm, _options) as VueWrapper<PopconfirmInstance>
  }
  const slots = { default: () => h('div', { id: 'trigger' }, 'trigger') }

  afterEach(() => {
    document.querySelector('.ix-popconfirm-overlay-container')!.innerHTML = ''
  })

  renderWork<PopconfirmProps>(Popconfirm, {
    props: { visible: true, title: 'Title' },
    slots,
    attachTo: 'body',
  })

  test('title work', async () => {
    const wrapper = PopconfirmMount({
      props: { visible: true, title: 'Title' },
      slots,
    })

    expect(document.querySelector('.ix-popconfirm-title')!.textContent).toBe('Title')

    await wrapper.setProps({ title: 'Title 2' })

    expect(document.querySelector('.ix-popconfirm-title')!.textContent).toBe('Title 2')
  })

  test('title slot work', async () => {
    PopconfirmMount({
      props: { visible: true, title: 'Title' },
      slots: { ...slots, title: () => h('div', 'Title slot') },
    })

    expect(document.querySelector('.ix-popconfirm-title')!.textContent).toBe('Title slot')
  })

  test('custom button work', async () => {
    let cancelText = 'No'
    const cancelButton = { mode: 'dashed' } as const
    let okText = 'Yes'
    const okButton = { mode: 'primary', danger: true } as const

    const wrapper = PopconfirmMount({ props: { cancelText, cancelButton, okText, okButton } })

    const [okButtonWrapper, cancelButtonWrapper] = await wrapper.getComponent(PopconfirmContent).findAll('.ix-button')

    expect(cancelButtonWrapper.text()).toBe(cancelText)
    expect(cancelButtonWrapper.classes()).toContain('ix-button-dashed')
    expect(okButtonWrapper.text()).toBe(okText)
    expect(okButtonWrapper.classes()).toContain('ix-button-primary')
    expect(okButtonWrapper.classes()).toContain('ix-button-danger')

    cancelText = 'No No'
    okText = 'Yes Yes'
    await wrapper.setProps({ cancelText, okText })

    const [okButtonWrapper1, cancelButtonWrapper1] = await wrapper.getComponent(PopconfirmContent).findAll('.ix-button')

    expect(cancelButtonWrapper1.text()).toBe(cancelText)
    expect(okButtonWrapper1.text()).toBe(okText)
  })

  test('onCancel work', async () => {
    const onCancel = vi.fn()
    const onUpdateVisible = vi.fn()
    const wrapper = PopconfirmMount({ props: { onCancel, 'onUpdate:visible': onUpdateVisible } })

    wrapper.vm.cancel()
    await flushPromises()

    expect(onCancel).toBeCalled()
    expect(onUpdateVisible).toBeCalledWith(false)
  })

  test('onCancel with result work', async () => {
    const onCancel = vi.fn().mockImplementation((evt: unknown) => evt === 'cancel')
    const onUpdateVisible = vi.fn()
    const wrapper = PopconfirmMount({ props: { onCancel, 'onUpdate:visible': onUpdateVisible } })

    wrapper.vm.cancel(1)
    await flushPromises()

    expect(onCancel).toBeCalledWith(1)
    expect(onUpdateVisible).not.toBeCalled()

    wrapper.vm.cancel('cancel')
    await flushPromises()

    expect(onCancel).toBeCalledWith('cancel')
    expect(onUpdateVisible).toBeCalledWith(false)
  })

  test('onCancel with promise work', async () => {
    const onCancel = vi.fn().mockImplementation((evt: unknown) => Promise.resolve(evt === 'cancel'))
    const onUpdateVisible = vi.fn()
    const wrapper = PopconfirmMount({ props: { onCancel, 'onUpdate:visible': onUpdateVisible } })

    wrapper.vm.cancel(1)
    await flushPromises()

    expect(onCancel).toBeCalledWith(1)
    expect(onUpdateVisible).not.toBeCalled()

    wrapper.vm.cancel('cancel')
    await flushPromises()

    expect(onCancel).toBeCalledWith('cancel')
    expect(onUpdateVisible).toBeCalledWith(false)
  })

  test('onOk work', async () => {
    const onOk = vi.fn()
    const onUpdateVisible = vi.fn()
    const wrapper = PopconfirmMount({ props: { onOk, 'onUpdate:visible': onUpdateVisible } })

    wrapper.vm.ok()
    await flushPromises()

    expect(onOk).toBeCalled()
    expect(onUpdateVisible).toBeCalledWith(false)
  })

  test('onOk with result work', async () => {
    const onOk = vi.fn().mockImplementation((evt: unknown) => evt === 'ok')
    const onUpdateVisible = vi.fn()
    const wrapper = PopconfirmMount({ props: { onOk, 'onUpdate:visible': onUpdateVisible } })

    wrapper.vm.ok(1)
    await flushPromises()

    expect(onOk).toBeCalledWith(1)
    expect(onUpdateVisible).not.toBeCalled()

    wrapper.vm.ok('ok')
    await flushPromises()

    expect(onOk).toBeCalledWith('ok')
    expect(onUpdateVisible).toBeCalledWith(false)
  })

  test('onOk with promise work', async () => {
    const onOk = vi.fn().mockImplementation((evt: unknown) => Promise.resolve(evt === 'ok'))
    const onUpdateVisible = vi.fn()
    const wrapper = PopconfirmMount({ props: { onOk, 'onUpdate:visible': onUpdateVisible } })

    wrapper.vm.ok(1)
    await flushPromises()

    expect(onOk).toBeCalledWith(1)
    expect(onUpdateVisible).not.toBeCalled()

    wrapper.vm.ok('ok')
    await flushPromises()

    expect(onOk).toBeCalledWith('ok')
    expect(onUpdateVisible).toBeCalledWith(false)
  })

  test('footer with boolean work', async () => {
    const wrapper = PopconfirmMount({ props: { footer: false } })
    const popconfirmWrapper = wrapper.getComponent(Popconfirm)

    expect(popconfirmWrapper.findComponent(PopconfirmContent).find('.ix-popconfirm-footer').exists()).toBe(false)

    await wrapper.setProps({ footer: undefined })

    expect(popconfirmWrapper.findComponent(PopconfirmContent).find('.ix-popconfirm-footer').exists()).toBe(true)
  })

  test('footer buttons work', async () => {
    let footer: PopconfirmButtonProps[] = [{ text: 'button1' }]
    const wrapper = PopconfirmMount({ props: { footer } })
    const popconfirmWrapper = wrapper.getComponent(Popconfirm)

    expect(popconfirmWrapper.findComponent(PopconfirmContent).findAll('.ix-button').length).toBe(1)

    footer = [...footer, { text: 'button2' }]
    await wrapper.setProps({ footer })

    expect(popconfirmWrapper.findComponent(PopconfirmContent).findAll('.ix-button').length).toBe(2)
  })

  test('footer slot work', async () => {
    const footer: PopconfirmButtonProps[] = [{ text: 'button1' }]
    const wrapper = PopconfirmMount({
      props: { footer },
      slots: { footer: () => h(IxButton, {}, { default: () => 'button slot' }) },
    })
    const popconfirmWrapper = wrapper.getComponent(Popconfirm)

    expect(popconfirmWrapper.findComponent(PopconfirmContent).findAll('.ix-button').length).toBe(1)
    expect(popconfirmWrapper.findComponent(PopconfirmContent).find('.ix-button').text()).toBe('button slot')
  })
})
