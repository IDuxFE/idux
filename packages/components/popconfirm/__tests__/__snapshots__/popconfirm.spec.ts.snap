// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popconfirm onCancel work 1`] = `"<p>Some contents...</p>"`;

exports[`Popconfirm render work 1`] = `
"<div id=\\"trigger\\">trigger</div>
<!--teleport start-->
<!--teleport end-->"
`;
